app/messenger/MainMessenger.js

```diff
@@ -106,8 +107,14 @@ class MainMessenger {
     this.contents = { ...props };
     const storage = new Storage();
     const install = storage.get('install');
-    if (!install) {
-      this.initializeInstallerWindow();
+    const wslConfigured = storage.get('wslConfigured');
+    let wsl2Supported = false;
+    if (isWindows) {
+      const build = os.release().split('.')[2];
+      wsl2Supported = Number(build) >= 19041;
+    }
+    if (!install || (isWindows && !wslConfigured && wsl2Supported)) {
```

Comment:
not sure `isWindows` is needed since `isWsl2Supported`  can only be true if isWindows is true
```suggestion
    if (!install || (!wslConfigured && isWsl2Supported)) {
```